rules/react.js

```diff
@@ -1,91 +1,146 @@
 module.exports = {
 	"rules": {
 		// Don't allow displayName parameters, but require the component to be named by reference.
-		"react/display-name": [2, {
-			"acceptTranspilerName": true
+		"react/display-name": ["error", {
+			"ignoreTranspilerName": false
 		}],
 		// Allow all prop types at this time.
-		"react/forbid-prop-types": 0,
+		"react/forbid-component-props": "off",
+		// Allow all element types.
+		"react/forbid-elements": "off",
+		// Don't allow messy imports of other component's prop types.
+		"react/forbid-foreign-prop-types": "error",
+		// Warn if the developer is using vague prop type descriptions, like any, array, and object.
+		"react/forbid-prop-types": "warn",
 		// Enforce boolean attributes notation No need for <Component variable={true} />.
 		// Just do <Component variable />.
-		"react/jsx-boolean-value": [2, "never"],
+		"react/jsx-boolean-value": ["error", "never"],
 		// Require closing brackets to be located immediately following the last prop.
-		"react/jsx-closing-bracket-location": [2, "after-props"],
+		"react/jsx-closing-bracket-location": ["error", "after-props"],
 		// Don't allow spaces between curly braces in JSX attributes.
-		"react/jsx-curly-spacing": [2, "never"],
+		"react/jsx-curly-spacing": ["error", "never", {
+			"allowMultiline": true
+		}],
+		// Don't allow spaces before or after the equal sign for components.
+		"react/jsx-equals-spacing": ["error", "never"],
+		// Allow either .js or .jsx files for components.
+		"react/jsx-filename-extension": ["warn", {
+			"extensions": [".js", ".jsx"]
+		}],
+		// The first prop of a component should never be on a new line.
+		"react/jsx-first-prop-new-line": ["error", "never"],
 		// Ensure correct prefixing of event handlers in JSX.
-		"react/jsx-handler-names": [1, {
+		"react/jsx-handler-names": ["warn", {
 			"eventHandlerPrefix": "handle",
 			"eventHandlerPropPrefix": "on"
 		}],
 		// Require a tab indentation in props.
-		"react/jsx-indent-props": [2, "tab"],
+		"react/jsx-indent-props": ["error", "tab"],
 		// Require tab indentations in all JSX components
-		"react/jsx-indent": [2, "tab"],
+		"react/jsx-indent": ["error", "tab"],
 		// Require key props to be used.
-		"react/jsx-key": 2,
+		"react/jsx-key": "error",
 		// Limit number of props on a single line in JSX to 1
-		"react/jsx-max-props-per-line":[2, {
-			"maximum": 1
+		"react/jsx-max-props-per-line":["error", {
+			"maximum": 1,
+			"when": "always"
 		}],
-		// Allow arrow functions and binding because "premature optimization is the root of all evil"
-		// or at least because it messes with HMR. (h/t Donald Knuth via Gabe Isman)
-		"react/jsx-no-bind": 0,
+		// Require binding to happen in the constructor.
+		"react/jsx-no-bind": ["error", {
+			"allowArrowFunctions": true
```

Comment:
We have been moving toward binding all functions in the constructor. We might have to make some changes for this to pass everywhere, but should be a small switch.